src/acl_kernel_if.cpp

```diff
@@ -846,8 +809,13 @@ int acl_kernel_if_init(acl_kernel_if *kern, acl_bsp_io bsp_io,
     return -1;
   }
 
-  result = acl_kernel_if_update(
-      sysdef->device[kern->physical_device_id].autodiscovery_def, kern);
+  if (acl_platform.offline_mode == ACL_CONTEXT_MPSIM) {
+    sysdef->device[kern->physical_device_id].autodiscovery_def.name =
```

Comment:
It seems like simulation still does some special handling here. Can you please add some comment to explain why the simulation cannot just call `acl_kernel_if_update` here?

Reply:
Hmm I think it has something to do with `simulation_set_kernel_cra_address_map` but I can't remember what exactly it was... Let me check again

Reply:
Added more logic in the simulation runtime and removed the special handling, now we just set device name and num_devices for sim.